hpe3parclient/client.py

```diff
@@ -4816,3 +4731,242 @@ def getScheduleStatus(self, schedule_name):
+                        'InstallTime': '2014-01-30 11:34:20 PST',
+                        'Id': 'DEVEL'
+                    },
+                    {
+                        'Package': 'Complete',
+                        'Version': '3.1.3.202',
+                        'InstallTime': '2014-03-26 13:59:42 PDT',
+                        'Id': 'GA'
+                    },
+                    {
+                        'Package': 'Complete',
+                        'Version': '3.1.3.230',
+                        'InstallTime': '2014-06-06 14:46:56 PDT',
+                        'Id': 'MU1'
+                    }
+                ]
+            }
+
+            # Example 2: No data rows, so zero members.
+            ret = {'total': 0, 'members': []}
+
+        """
+
+        members = []
+        if cli_output and len(cli_output) >= 2:
+            for index, line in enumerate(cli_output):
+                if index == 0:
+                    headers = line.split(',')
+                else:
+                    split = line.split(',')
+                    member = {}
+                    for i, header in enumerate(headers):
+                        try:
+                            member[header] = split[i]
+                        except IndexError:
+                            member[header] = None
+                    members.append(member)
+
+        return {'total': len(members), 'members': members}
+
+    @staticmethod
+    def getPortNumber(ip, login, password, port=22,
+                      conn_timeout=None, privatekey=None,
+                      **kwargs):
+        """Get port number from showwsapi output
+
+        :param 3PAR credentials
+        :return: HTTPS_Port column value
+        """
+        ssh_client = ssh.HPE3PARSSHClient(ip, login, password, port,
+                                          conn_timeout, privatekey,
+                                          **kwargs)
+        if ssh_client is None:
+            raise exceptions.SSHException('SSH is not initialized. Initialize'
+                                          ' it by calling "setSSHOptions".')
+        ssh_client.open()
+        cli_output = ssh_client.run(['showwsapi'])
+        if ssh_client:
+            ssh_client.close()
+        wsapi_collection = HPE3ParClient.convert_cli_output_to_wsapi_format(
+            cli_output)
+        return wsapi_collection['members'][0]['HTTPS_Port']
+
+    def tuneVolume(self, volName, tune_operation, optional=None):
+        """Tune a volume.
+
+        :param name: the name of the volume
+        :type name: str
+        :param name: tune_operation 1 for USR_CPG 2 for SNP_CPG
+        :type name: int
+        :param optional: dictionary of volume attributes to change
+        :type optional: dict
+        .. code-block:: python
+
+            optional = {
+             'action': 6,                  # For tuneVolume operation
+             'tuneOperation': 1,           # 1 for USR_CPG,2 for SNP_CPG
```

Comment:
Why are we expecting 'tuneOperation' in dict when this function accepts tune_operation as a parameter? Do we need to update the documentation string?

Reply:
> Why are we expecting 'tuneOperation' in dict when this function accepts tune_operation as a parameter? Do we need to update the documentation string?

Addressed this comment